src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -436,6 +451,91 @@ public String call() throws Exception {
     return response;
   }
 
+  private String putRequest(String path, String status) throws UnsupportedEncodingException {
+    logger.log(Level.FINEST, "PR-PUT-REQUEST:" + path + " with: " + status);
+    HttpClient client = getHttpClient();
+    client.getState().setCredentials(AuthScope.ANY, credentials);
+
+    PutMethod httpput = new PutMethod(path);
+    httpput.setRequestHeader("Connection", "close");
+    httpput.setRequestHeader("X-Atlassian-Token", "no-check"); // xsrf
+
+    if (status != null) {
+      ObjectNode node = mapper.getNodeFactory().objectNode();
+      node.put("status", status);
+      StringRequestEntity requestEntity = null;
+      try {
+        requestEntity =
+            new StringRequestEntity(mapper.writeValueAsString(node), "application/json", "UTF-8");
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      httpput.setRequestEntity(requestEntity);
+    }
+
+    String response = "";
+    FutureTask<String> httpTask = null;
+    Thread thread;
+
+    try {
+      httpTask =
```

Comment:
This is some bad code. Unfortunately it's a copy paste from other methods for doing http requests. 
Not requesting it for this PR but we definitely need to get rid of this thread spawning.